tools/toolchains/iar.py

```diff
@@ -126,17 +126,22 @@ def cc_extra(self, object):
         base, _ = splitext(object)
         return ["-l", base + '.s.txt']
 
-    def get_compile_options(self, defines, includes):
+    def get_compile_options(self, defines, includes, for_asm=False):
         opts = ['-D%s' % d for d in defines] + ['-f', self.get_inc_file(includes)]
-        config_header = self.get_config_header()
-        if config_header is not None:
-            opts = opts + ['--preinclude', config_header]
+        if for_asm:
+            # The assembler doesn't support '--preinclude', so we need to add
+            # the macros directly
+            opts = opts + ['-D%s' % d for d in self.get_config_macros()]
```

Comment:
> you won't get an error until you try to use that macro

A tool like uvision can't go further with assembly defines as shown above, therefore ``-DWRONG_MACRO={1,2,3,4}`` returns error that parsing failed (it's slightly cryptic error). That's what I experienced, therefore there's a fix in mbed codebase to remove floating point from TIMESTAMP for the macros in the uvision exporter.